CLN: Enigma DataReader

Following https://github.com/pydata/pandas-datareader/pull/245

Some code cleanup is required
see https://travis-ci.org/pydata/pandas-datareader/jobs/166965935

Pinging @trevorprater @gregorykremler @m1teyssier

```
$ flake8 --ignore E501 pandas_datareader
pandas_datareader/base.py:4:1: F401 'datetime as dt' imported but unused
pandas_datareader/base.py:7:1: F401 'requests_file.FileAdapter' imported but unused
pandas_datareader/base.py:9:1: F401 'pandas.to_datetime' imported but unused
pandas_datareader/base.py:11:1: F401 'pandas.core.common.is_number' imported but unused
pandas_datareader/enigma.py:2:1: F401 'json' imported but unused
pandas_datareader/enigma.py:4:1: F401 'sys' imported but unused
pandas_datareader/enigma.py:8:1: F401 'pandas.DataFrame' imported but unused
pandas_datareader/enigma.py:45:20: E711 comparison to None should be 'if cond is None:'
pandas_datareader/enigma.py:47:30: E711 comparison to None should be 'if cond is None:'
pandas_datareader/enigma.py:59:1: W293 blank line contains whitespace
pandas_datareader/enigma.py:60:5: E303 too many blank lines (2)
pandas_datareader/enigma.py:69:1: W293 blank line contains whitespace
pandas_datareader/enigma.py:70:5: E303 too many blank lines (2)
pandas_datareader/enigma.py:74:1: W293 blank line contains whitespace
pandas_datareader/enigma.py:75:5: E303 too many blank lines (2)
pandas_datareader/enigma.py:81:1: W293 blank line contains whitespace
pandas_datareader/enigma.py:82:5: E303 too many blank lines (2)
pandas_datareader/enigma.py:85:1: W293 blank line contains whitespace
pandas_datareader/enigma.py:86:5: E303 too many blank lines (2)
pandas_datareader/tests/test_enigma.py:3:1: F401 'requests' imported but unused
pandas_datareader/tests/test_enigma.py:8:1: F401 'pandas.util.testing.assert_series_equal' imported but unused
pandas_datareader/tests/test_enigma.py:8:1: F401 'pandas.util.testing.assert_frame_equal' imported but unused
```

Could you open an other PR to fix this ?

--- a/.travis.yml
+++ b/.travis.yml
@@ -4,20 +4,11 @@ language: python
 
 env:
   - PYTHON=2.7 PANDAS=0.16.2
+  - PYTHON=2.7 PANDAS=0.17.1
+  - PYTHON=2.7 PANDAS=0.19.0
   - PYTHON=3.4 PANDAS=0.17.1
   - PYTHON=3.4 PANDAS=0.18.1
-  - PYTHON=3.5 PANDAS=0.18.1
-  - PYTHON=2.7 PANDAS=0.13.0
-  - PYTHON=2.7 PANDAS=0.12.0
-  - PYTHON=2.7 PANDAS=0.11.0
-
-matrix:
-    allow_failures:
-        - env: PYTHON=2.6 PANDAS=0.14.1
-        - env: PYTHON=2.7 PANDAS=0.13.0
-        - env: PYTHON=2.7 PANDAS=0.12.0
-        - env: PYTHON=2.7 PANDAS=0.11.0
-        - env: PYTHON=3.3 PANDAS=0.15.1
+  - PYTHON=3.5 PANDAS=0.19.0
 
 install:
   - pip install -qq flake8

--- a/pandas_datareader/_utils.py
+++ b/pandas_datareader/_utils.py
@@ -2,7 +2,7 @@ import datetime as dt
 from distutils.version import LooseVersion
 
 import pandas as pd
-from pandas.core.common import PandasError, is_number
+from pandas.core.common import PandasError
 from pandas import to_datetime
 
 import requests
@@ -15,6 +15,30 @@ if pd.compat.PY3:
     from urllib.error import HTTPError     # noqa
 else:
     from urllib2 import HTTPError          # noqa
+
+PANDAS_VERSION = LooseVersion(pd.__version__)
+
+if PANDAS_VERSION >= LooseVersion('0.19.0'):
+    PANDAS_0190 = True
+    from pandas.api.types import is_number                   # noqa
+else:
+    PANDAS_0190 = False
+    from pandas.core.common import is_number                 # noqa
+
+if PANDAS_VERSION >= LooseVersion('0.17.0'):
+    PANDAS_0170 = True
+else:
+    PANDAS_0170 = False
+
+if PANDAS_VERSION >= LooseVersion('0.16.0'):
+    PANDAS_0160 = True
+else:
+    PANDAS_0160 = False
+
+if PANDAS_VERSION >= LooseVersion('0.14.0'):
+    PANDAS_0140 = True
+else:
+    PANDAS_0140 = False
 
 
 class SymbolWarning(UserWarning):
@@ -53,20 +77,3 @@ def _init_session(session, retry_count=3):
         session.mount('file://', FileAdapter())
         # do not set requests max_retries here to support arbitrary pause
     return session
-
-PANDAS_VERSION = LooseVersion(pd.__version__)
-
-if PANDAS_VERSION >= LooseVersion('0.17.0'):
-    PANDAS_0170 = True
-else:
-    PANDAS_0170 = False
-
-if PANDAS_VERSION >= LooseVersion('0.16.0'):
-    PANDAS_0160 = True
-else:
-    PANDAS_0160 = False
-
-if PANDAS_VERSION >= LooseVersion('0.14.0'):
-    PANDAS_0140 = True
-else:
-    PANDAS_0140 = False

--- a/pandas_datareader/base.py
+++ b/pandas_datareader/base.py
@@ -1,14 +1,10 @@
 import time
 import warnings
 import numpy as np
-import datetime as dt
 
 import requests
-from requests_file import FileAdapter
 
-from pandas import to_datetime
 import pandas.compat as compat
-from pandas.core.common import is_number
 from pandas import Panel, DataFrame
 from pandas import read_csv
 from pandas.io.common import urlencode

--- a/pandas_datareader/enigma.py
+++ b/pandas_datareader/enigma.py
@@ -1,11 +1,8 @@
 import zlib
-import json
 import os
-import sys
 import time
 from pandas.compat import StringIO
 
-from pandas import DataFrame
 import pandas.compat as compat
 import pandas as pd
 import requests
@@ -42,9 +39,9 @@ class EnigmaReader(_BaseReader):
         super(EnigmaReader, self).__init__(symbols=[],
                                            retry_count=retry_count,
                                            pause=pause)
-        if api_key == None:
+        if api_key is None:
             self._api_key = os.getenv('ENIGMA_API_KEY')
-            if self._api_key == None:
+            if self._api_key is None:
                 raise ValueError(
                     """Please provide an Enigma API key or set the ENIGMA_API_KEY environment variable\n
                         If you do not have an API key, you can get one here: https://app.enigma.io/signup""")
@@ -56,7 +53,6 @@ class EnigmaReader(_BaseReader):
             raise ValueError(
                 "The Enigma datapath must be a string (ex: 'enigma.inspections.restaurants.fl')")
 
-            
     @property
     def url(self):
         return 'https://api.enigma.io/v2/export/{}/{}'.format(self._api_key,
@@ -66,23 +62,19 @@ class EnigmaReader(_BaseReader):
     def export_key(self):
         return 'export_url'
 
-    
     @property
     def _head_key(self):
         return 'head_url'
 
-    
     def _request(self, url):
         self.session.headers.update({'User-Agent': 'pandas-datareader'})
         resp = self.session.get(url)
         resp.raise_for_status()
         return resp
 
-    
     def _decompress_export(self, compressed_export_data):
         return zlib.decompress(compressed_export_data, 16 + zlib.MAX_WBITS)
 
-    
     def extract_export_url(self, delay=10, max_attempts=10):
         """
         Performs an HTTP HEAD request on 'head_url' until it returns a `200`.

--- a/pandas_datareader/tests/test_edgar.py
+++ b/pandas_datareader/tests/test_edgar.py
@@ -1,4 +1,6 @@
 import nose
+
+import pandas as pd
 import pandas.util.testing as tm
 
 import pandas_datareader.data as web
@@ -7,13 +9,15 @@ from pandas_datareader._utils import RemoteDataError
 
 class TestEdgarIndex(tm.TestCase):
 
-    def setUp(self):
-        raise nose.SkipTest()
-
     def test_get_full_index(self):
         try:
             ed = web.DataReader('full', 'edgar-index')
             assert len(ed) > 1000
+
+            exp_columns = pd.Index(['cik', 'company_name', 'form_type',
+                                    'date_filed', 'filename'], dtype='object')
+            tm.assert_index_equal(ed.columns, exp_columns)
+
         except RemoteDataError as e:
             raise nose.SkipTest(e)
 
@@ -22,10 +26,21 @@ class TestEdgarIndex(tm.TestCase):
             ed = web.DataReader('daily', 'edgar-index', '1994-06-30',
                                 '1994-07-02')
             assert len(ed) > 200
+
+            self.assertEqual(ed.index.nlevels, 2)
+            dti = ed.index.get_level_values(0)
+            self.assertIsInstance(dti, pd.DatetimeIndex)
+            exp_columns = pd.Index(['company_name', 'form_type',
+                                    'filename'], dtype='object')
+            tm.assert_index_equal(ed.columns, exp_columns)
+
         except RemoteDataError as e:
             raise nose.SkipTest(e)
 
     def test_get_gz_index_and_no_date(self):
+        # the test causes Travis timeout
+        raise nose.SkipTest()
+
         try:
             ed = web.DataReader('daily', 'edgar-index')
             assert len(ed) > 2000
@@ -34,11 +49,23 @@ class TestEdgarIndex(tm.TestCase):
 
     def test_6_digit_date(self):
         try:
-            ed = web.DataReader('daily', 'edgar-index', '1998-05-18',
-                                '1998-05-18')
+            ed = web.DataReader('daily', 'edgar-index', start='1998-05-18',
+                                end='1998-05-18')
             assert len(ed) < 1200
+
+            self.assertEqual(ed.index.nlevels, 2)
+            dti = ed.index.get_level_values(0)
+            self.assertIsInstance(dti, pd.DatetimeIndex)
+            self.assertEqual(dti[0], pd.Timestamp('1998-05-18'))
+            self.assertEqual(dti[-1], pd.Timestamp('1998-05-18'))
+
+            exp_columns = pd.Index(['company_name', 'form_type',
+                                    'filename'], dtype='object')
+            tm.assert_index_equal(ed.columns, exp_columns)
+
         except RemoteDataError as e:
             raise nose.SkipTest(e)
+
 
 if __name__ == '__main__':
     nose.runmodule(argv=[__file__, '-vvs', '-x', '--pdb', '--pdb-failure'],

--- a/pandas_datareader/tests/test_edgar.py
+++ b/pandas_datareader/tests/test_edgar.py
@@ -6,6 +6,10 @@ from pandas_datareader._utils import RemoteDataError
 
 
 class TestEdgarIndex(tm.TestCase):
+
+    def setUp(self):
+        raise nose.SkipTest()
+
     def test_get_full_index(self):
         try:
             ed = web.DataReader('full', 'edgar-index')

--- a/pandas_datareader/tests/test_enigma.py
+++ b/pandas_datareader/tests/test_enigma.py
@@ -1,11 +1,9 @@
 import os
 
-import requests
 from requests.exceptions import HTTPError
 
 import nose
 import pandas.util.testing as tm
-from pandas.util.testing import (assert_series_equal, assert_frame_equal)
 from pandas_datareader.tests._utils import _skip_if_no_lxml
 
 import pandas_datareader.data as web
@@ -15,10 +13,14 @@ TEST_API_KEY = os.getenv('ENIGMA_API_KEY')
 
 
 class TestEnigma(tm.TestCase):
+
     @classmethod
     def setUpClass(cls):
         super(TestEnigma, cls).setUpClass()
         _skip_if_no_lxml()
+
+    def setUp(self):
+        raise nose.SkipTest()
 
     def test_enigma(self):
         self.assertTrue('serialid' in list(

--- a/pandas_datareader/tests/test_famafrench.py
+++ b/pandas_datareader/tests/test_famafrench.py
@@ -9,6 +9,10 @@ from pandas_datareader.tests._utils import _skip_if_no_lxml
 
 
 class TestFamaFrench(tm.TestCase):
+
+    def setUp(self):
+        raise nose.SkipTest()
+
     def test_get_data(self):
         keys = [
             'F-F_Research_Data_Factors', 'F-F_ST_Reversal_Factor',

--- a/pandas_datareader/tests/test_famafrench.py
+++ b/pandas_datareader/tests/test_famafrench.py
@@ -10,9 +10,6 @@ from pandas_datareader.tests._utils import _skip_if_no_lxml
 
 class TestFamaFrench(tm.TestCase):
 
-    def setUp(self):
-        raise nose.SkipTest()
-
     def test_get_data(self):
         keys = [
             'F-F_Research_Data_Factors', 'F-F_ST_Reversal_Factor',
@@ -21,18 +18,18 @@ class TestFamaFrench(tm.TestCase):
         ]
         for name in keys:
             ff = web.DataReader(name, 'famafrench')
-            assert 'DESCR' in ff
-            assert len(ff) > 1
+            self.assertTrue('DESCR' in ff)
+            self.assertTrue(len(ff) > 1)
 
     def test_get_available_datasets(self):
         _skip_if_no_lxml()
         l = get_available_datasets()
-        assert len(l) > 100
+        self.assertTrue(len(l) > 100)
 
     def test_index(self):
         ff = web.DataReader('F-F_Research_Data_Factors', 'famafrench')
-        assert ff[0].index.freq == 'M'
-        assert ff[1].index.freq == 'A-DEC'
+        self.assertEqual(ff[0].index.freq, 'M')
+        self.assertEqual(ff[1].index.freq, 'A-DEC')
 
     def test_f_f_research(self):
         results = web.DataReader("F-F_Research_Data_Factors", "famafrench",
@@ -42,10 +39,10 @@ class TestFamaFrench(tm.TestCase):
 
         exp = pd.DataFrame({'Mkt-RF': [-3.36, 3.4, 6.31, 2., -7.89, -5.56,
                                        6.93, -4.77, 9.54, 3.88, 0.6, 6.82],
-                            'SMB': [0.2, 1.44, 1.57, 4.92, -0.09, -2.15,
-                                    0.24, -3.03, 3.84, 1.01, 3.69, 0.85],
-                            'HML': [0.61, 2.74, 2.01, 3.12, -2.32, -4.27,
-                                    0.04, -1.51, -2.94, -2.23, -0.58, 3.47],
+                            'SMB': [0.37, 1.19, 1.49, 4.99, 0.0, -2.01, 0.21,
+                                    -2.99, 3.92, 1.14, 3.68, 0.68],
+                            'HML': [0.3, 3.18, 2.15, 2.83, -2.41, -4.52, -0.21,
+                                    -1.96, -3.12, -2.52, -0.91, 3.78],
                             'RF': [0., 0., 0.01, 0.01, 0.01, 0.01, 0.01,
                                    0.01, 0.01, 0.01, 0.01, 0.01]},
                            index=pd.period_range('2010-01-01', '2010-12-01', freq='M', name='Date'),

--- a/pandas_datareader/tests/test_google_options.py
+++ b/pandas_datareader/tests/test_google_options.py
@@ -47,6 +47,7 @@ class TestGoogleOptions(tm.TestCase):
             self.goog.get_options_data(month=1, year=2016)
 
     def test_expiry_dates(self):
+        raise nose.SkipTest()
         try:
             dates = self.goog.expiry_dates
         except RemoteDataError as e:  # pragma: no cover

--- a/pandas_datareader/tests/test_google_options.py
+++ b/pandas_datareader/tests/test_google_options.py
@@ -1,5 +1,7 @@
 import nose
 
+from datetime import date
+
 import numpy as np
 import pandas as pd
 import pandas.util.testing as tm
@@ -47,12 +49,14 @@ class TestGoogleOptions(tm.TestCase):
             self.goog.get_options_data(month=1, year=2016)
 
     def test_expiry_dates(self):
-        raise nose.SkipTest()
         try:
             dates = self.goog.expiry_dates
         except RemoteDataError as e:  # pragma: no cover
             raise nose.SkipTest(e)
-        self.assertTrue(len(dates) > 6)
+
+        self.assertTrue(len(dates) >= 5)
+        self.assertIsInstance(dates, list)
+        self.assertTrue(all(isinstance(dt, date) for dt in dates))
 
     def test_get_call_data(self):
         with tm.assertRaises(NotImplementedError):

